### PR TITLE
[HttpKernel] tweaked redirection profiling in RequestDataCollector

### PR DESCRIPTION
**Pull request by HeahDude**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | ~ |
| License | MIT |
| Doc PR | ~ |
- c8ba3b2 removes duplicated code forgotten in #17589
- a47d2e8 fixes the collecting of redirect data in first sub request instead of redirected master request.

### CODE REVIEW

**Comment by HeahDude** [comment]
@javiereguiluz it happens because of this line https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/EventListener/ProfilerListener.php#L81.

But the scope of `RequestCollector::collect()` does not allow to know it the request is the sub request.
So I had to register the `RequestDataCollector` on `KernelEvents::RESPONSE` to get it from the event.

I also refactored a bit since `$session` was already defined in `RequestDataCollector::collect()`.

Please tell me if the fix works for you too, and thanks again for reporting it :)

**Comment by HeahDude** [comment]
I reverted the redirect controller thing as I got the wrong while testing it with `symfony-demo`.
Also I've pushed df19c14 to store the redirected flag in the request attributes instead of the session and used the returned value of `Session::remove`. If you agree with this commit I'll quash it.

Thanks.

**Comment by fabpot** [comment]
ping @javiereguiluz

**Comment by fabpot** [comment]
Thank you @HeahDude.

File: src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php [hunk 1]
```diff
@@ -128,33 +129,22 @@ public function collect(Request $request, Response $response, \Exception $except
...
-            $request->getSession()->set('sf_redirect', array(
-                'token' => $response->headers->get('x-debug-token'),
-                'route' => $request->attributes->get('_route', 'n/a'),
-                'method' => $request->getMethod(),
-                'controller' => $this->parseController($request->attributes->get('_controller')),
-                'status_code' => $statusCode,
-                'status_text' => Response::$statusTexts[(int) $statusCode],
-            ));
-        }
+        if (isset($session)) {
```
Comment (fabpot) [inline src/Symfony/Component/HttpKernel/DataCollector/RequestDataCollector.php]: Using `isset` on a variable should be avoided. I've fixed it in becdbd9ea8fc43c0400177eb6306d5aad63989ad

**Pull request merged**